### PR TITLE
Update meld.rb

### PR DESCRIPTION
**Pull request by peterwwillis**
Meld fork is under active development

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [ ] `brew audit --cask --online <cask>` is error-free.
- [ ] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [ ] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [ ] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

**Comment by bevanjkay** [comment]
@peterwwillis The app has not received a release in 4 years, which is why it has been deprecated here.

**Comment by peterwwillis** [comment]
@bevanjkay does Homebrew have a governance policy you can link me to which provides the criteria for being considered deprecated? because I'm not sure it's warranted...

 - You can see that the Meld project [does have new releases](https://gitlab.gnome.org/GNOME/meld/-/releases)
 - Officially Mac OS is not supported by the Meld project, so a volunteer has forked it and done the work of [porting and packaging it for Mac OS](https://github.com/yousseb/meld/releases). 
   - @yousseb has been the sole maintainer of this fork. Despite the fact that his brother was [murdered in a terrorist attack](https://www.stuff.co.nz/national/christchurch-shooting/111375038/osama-abu-kwaik-was-a-refugee-born-to-an-orphan-who-died-in-christchurch-the-city-he-loved), he continued to work on the fork for another year, after which his work slowed.
   - In the mean time, the project continued to be used, and pull requests and issues opened.
     - Closed PRs: [131](https://github.com/yousseb/meld/pull/131), [137](https://github.com/yousseb/meld/pull/137), [139](https://github.com/yousseb/meld/pull/139), [140](https://github.com/yousseb/meld/pull/140), [144](https://github.com/yousseb/meld/pull/144)
     - Open PRs: [145](https://github.com/yousseb/meld/pull/145), [148](https://github.com/yousseb/meld/pull/148), [150](https://github.com/yousseb/meld/pull/150)
     - Open Issues (88): https://github.com/yousseb/meld/issues?q=is%3Aopen+is%3Aissue
  - A pre-release was issued a few months ago: https://github.com/yousseb/meld/releases/tag/osx-20


It's also worth noting that [Homebrew's designation as deprecated](https://github.com/yousseb/meld/issues/143#issuecomment-1503408235) might make people [think the project is dead](https://gitlab.gnome.org/GNOME/meld/-/issues/804), which makes them not want to work on it or use it.

Now, imagine you're just a guy using a computer. You have a family and responsibilities. Occasionally you write code, and you use a useful tool for it, but notice it isn't available for your platform. So you take the time (years) to get it working, and release it. You don't have a lot of time to spend maintaining it, so time passes with few releases. But you still keep track of what's going on and try to do a bit of work now and then. Later, a different popular OSS project decides you haven't done enough, and that your work is now deprecated (even though you do respond to issues, PRs, and get pre-releases ready). So now new people are discouraged from your work (because why help or use a deprecated project?)

Clearly, [some people](https://github.com/yousseb/meld/issues/149) [want to continue the project](https://github.com/yousseb/meld/issues/155). But that isn't helped by marking it deprecated. I'll quote some of that writing:

> First of all, bearing in mind this repo's README has come straight from the upstream, please make sure you have read yousseb's own project home page at https://yousseb.github.io/meld/ – there you will get an understanding not only of how to best use the software in its current state, but also of how the maintainer sees this project in relation to the upstream. Also see [the releases](https://github.com/yousseb/meld/releases), showing not only that this project is definitely not dead, and that yousseb has done well to keep within just one feature version of the upstream. It also indicates ares where the community might be able to help out to improve packages and/or functionality.

> So, where are we now? https://github.com/yousseb/meld/issues/143 asks if there is a plan for new version, yet [release osx-20](https://github.com/yousseb/meld/releases/tag/osx-20) shows that there now clearly is. [Brew's reports of discontinuation](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/m/meld.rb) now seem a little unfair.

I echo the comment that the deprecation seems unfair.

There have been proposals to fork the main project, and fork the fork. There have been proposals to try to upstream the port (that won't happen for some time, upstream is busy with a GTK4 port). There have been proposals to take over this port, but only if a single new maintainer gets total control of it.

Despite all of this, and despite the lack of constant work on this one port, the version supported by this port is only one minor version away from the project's official releases. The project continues to have PRs and issues, and users continue to help each other use the software through this port's github project.

If you would like to encourage open source development and usage, rather than discourage it, I recommend removing the deprecation warning. There has been a pre-release made recently, and there is activity right now which may lead to another full release.

I would also add that it would be nice to display a note that this project is in need of volunteers. All it takes is one person seeing that notice, with enough time, expertise, and passion to put in the work needed to keep up with Mac OS's constant breaking changes. But if all they see is "deprecated", they may not bother at all.

**Comment by MikeMcQuaid** [comment]
> Now, imagine you're just a guy using a computer. You have a family and responsibilities.

@peterwwillis This applies to most Homebrew maintainers, too, and I don't think you're cutting us as much slack here.

> * Officially Mac OS is not supported by the Meld project

This alone would be enough for it to be deprecated.

> * A pre-release was issued a few months ago

But not a stable release, which Homebrew requires.

> There have been proposals to try to upstream the port (that won't happen for some time, upstream is busy with a GTK4 port).

If this happens: it can be undeprecated.

> But if all they see is "deprecated", they may not bother at all.

This somewhat contradicts your previous points about this being a very active, maintained project.

Look: yousseb, perhaps yourself and many other people have and continue to do great work in open source software. That, however, does not mean we should change our policies in Homebrew just because other people are working hard. This deprecation is not a critique on their work, even if you take it that way, it's a recognition of our packaging process.

We are also a project run by volunteers. At last estimate we have around a million users per maintainer. We have our policies so this project can be sustainable and not die (which has been closer than anyone realises in the past).

TL;DR: sorry but: passing on this PR.
